### ISSUE TITLE
consider filename changes for src/core

### ISSUE THREAD

**Issue by vijithassar** [processing/p5.js#2947]
#### Nature of issue?

- [x] Existing feature enhancement

#### Most appropriate sub-area of p5.js?

- [x] Core/Environment/Rendering

Opening this issue as part of my research for my Processing Foundation Fellowship project working on internal developer-facing documentation and principles.

I think it's worth considering changing these filenames in `src/core` in the interest of clarity; I've articulated separate reasoning for each. I'm posting them all here in one issue, but it may be wise to break out into individual issues per filename for more detailed discussion.

- [ ] `2d_primitives.js` feels a bit wordy, especially because there is no file that defines 3d primitives, and no file that defines other non-drawing primitives. This is mostly about drawing shapes, and could be `shapes.js` or similar.
- [ ] `attributes.js` strongly suggests DOM attributes, not the miscellaneous drawing parameters it actually contains.
- [ ] `canvas.js` doesn't really contain functionality that is directly related to `<canvas>` elements.
- [ ] `core.js` is a bit redundant. This entire repository can be considered the p5 "core" (as distinct from addons), and it already has the `src/core` directory, and within that directory is `src/core/core.js`.
- [x] `structure.js` doesn't structure the whole library or the p5 *object*; it could be argued that it structures the drawing loop, but I think something that refers to that using a less generic term (maybe `loop.js`, if nothing else?) would be helpful.
- [ ] `p5.Graphics.js` creates optional graphics *buffers* for bespoke rendering scenarios, but certainly does not contain all graphics-related functionality.
- [x] `api_notifications.js`: "api" and "notifications" are both broad terms that apply to other concepts in this codebase — the p5 object provides a public API, Friendly Error Messages can be considered notifications, and so on. Maybe `legacy.js` or `processingCompatibility.js` or similar?

**Comment by lmccart** [comment]
> - `2d_primitives.js` feels a bit wordy, especially because there is no file that defines 3d primitives, and no file that defines other non-drawing primitives. This is mostly about drawing shapes, and could be `shapes.js` or similar.
> - `attributes.js` strongly suggests DOM attributes, not the miscellaneous drawing parameters it actually contains.
Both these filenames refer to the subheading under the category "Shapes".
![screenshot of reference showing these subheadings](https://i.imgur.com/bGLJ5UI.png)
There is actually a 3d primitives file in the `src/webgl/` folder, but it is just named `primitives.js`. Given this, would it make sense to prefix both of these with "shape_" to be more clear? ie `shape_2d_primitives.js` and `shape_attributes.js`? It is wordier though..

> - `canvas.js` doesn't really contain functionality that is directly related to `<canvas>` elements.
Agree. I think what this is is a file analogous to `constants.js` but with helper functions. `helpers.js` or `shape_helpers.js`?

> - `core.js` is a bit redundant. This entire repository can be considered the p5 "core" (as distinct from addons), and it already has the `src/core` directory, and within that directory is `src/core/core.js`.
Do you have any suggestion or intuition here? `main.js`?

> - `structure.js` doesn't structure the whole library or the p5 *object*; it could be argued that it structures the drawing loop, but I think something that refers to that using a less generic term (maybe `loop.js`, if nothing else?) would be helpful.
"structure" here refers to the top level reference category, in the same way as "constants" and "environment", so maybe this one can stay?

> - `p5.Graphics.js` creates optional graphics *buffers* for bespoke rendering scenarios, but certainly does not contain all graphics-related functionality.
This one is referring to the object that is contained within it.. following the pattern of p5.Font.js, p5.Image.js, etc. So I think this makes sense? Or are you suggesting the object name itself should change?

> - `api_notifications.js`: "api" and "notifications" are both broad terms that apply to other concepts in this codebase — the p5 object provides a public API, Friendly Error Messages can be considered notifications, and so on. Maybe `legacy.js` or `processingCompatibility.js` or similar?
Agree. I like `legacy.js` since it could then also allow eventual space for deprecated function warns if needed.

**Comment by vijithassar** [comment]
- I agree, `main.js` would make sense.
- I suspect that renaming the `p5.Graphics` object (and consequently also the filename) might be in order, but I'll defer to others on this question
- `shape_helpers.js` or similar would be a nice description, but also maybe that could be combined with `2d_primitives.js` in a file named `shapes.js`?

**Comment by lmccart** [comment]
ok some decisions after some discussion at the p5.js:
* `canvas.js` > `helpers.js`
* `core.js` > `main.js`
* `webgl/primitives` > `webgl/3d_primitives.js`
* move `2d_primitive.js`, `attirbutes.js`, `curves.js`, and `vertex.js` to `shape` folder within `core` folder

@kjhollen sound ok?
